packages/core/src/common/keybinding.ts

```diff
@@ -267,4 +303,37 @@ export class KeybindingRegistry {
     isPseudoCommand(commandId: string): boolean {
         return commandId === KeybindingRegistry.PASSTHROUGH_PSEUDO_COMMAND;
     }
+
+    setKeymap(scope: number, rawKeyBindings: RawKeybinding[]) {
+        const customBindings: Keybinding[] = [];
+        for (const rawKeyBinding of rawKeyBindings) {
+            if (this.commandRegistry.getCommand(rawKeyBinding.command)) {
+                const code = KeyCode.parse(rawKeyBinding.keybinding);
+                if (code) {
+                    let context: KeybindingContext | undefined;
+                    if (rawKeyBinding.context) {
+                        context = this.contextRegistry.getContext(rawKeyBinding.context);
+                    }
+
+                    customBindings.push({
+                        commandId: rawKeyBinding.command,
+                        keyCode: code,
+                        contextId: context ? context.id : undefined
+                    });
+
+                } else {
+                    this.resetKeybindings(scope);
+                    return;
+                }
+            } else {
+                this.resetKeybindings(scope);
```

Comment:
Add user warning here?

Reply:
Also this only means to that the user tried to define a wrong command... is it worth resetting in that case ? 
I understand you want to do that if the whole keymap fails to parse but for individual bindings I think we should just warn the user ?

Reply:
I agree, I'll log an error for now.